Debug and improvements paired today (see commit msg)

mfranck opened the pull request.

mfranck committed:
Getting standard output in bucos:run/1 even in case of errors.
Added bucinet:active_ips/0,
Filtered "vbox" fake net interfaces out of "active ips"
Debug: Path expansion added in buccode:priv_dir/1

glejeune commented on src/bucos.erl:
```diff
@@ -10,7 +10,7 @@ run(Cmd) ->
 % Execute the given shell command
 % @end
--spec run(string() | binary(), integer()) -> {ok, term()} | {error, term()}.
+-spec run(string() | binary(), integer()) -> {ok, term()} | {error, term()} | {error, term(), string()}.
```
You break the backward compatibility. Maybe you need to create a new function : 

```
-spec run(string() | binary(), [options()]) -> {ok, term()} | {error, term()} | {error, term(), string()}.
```

Where `options()` is a list with the following options :
- `{timeout, integer()}` : for timeout
- `stdout_on_error` : to return `{error, integer(), string()}` on error (`{error, integer()}` otherwise)

mfranck commented:
Yep, I didn't pattern-matched it !

mfranck committed:
Corrected name of mispelled functions + tests added

mfranck committed:
Retro-compatibility for "bucos:run/1",
providing an interface that takes an "options" list in input,
leaving unchanged former interfaces and returned values.
